src/main/java/uk/gov/hmcts/darts/cases/service/impl/CaseServiceImpl.java

```diff
@@ -130,5 +133,18 @@ public List<AdvancedSearchResult> advancedSearch(GetCasesSearchRequest request)
         return AdvancedSearchResponseMapper.mapResponse(hearings);
     }
 
-
+    public List<EventResponse> getEvents(Integer caseId, Integer hearingId) {
+        List<Hearing> hearings = getCaseHearings(caseId);
+        if (hearings.isEmpty()) {
+            throw new DartsApiException(CaseApiError.CASE_NOT_FOUND);
+        }
+        Hearing hearing = hearings
+            .stream()
+            .filter(hearingEntity -> hearingEntity.getId().equals(hearingId))
+            .findFirst().orElseThrow(
+                () -> new DartsApiException(CaseApiError.HEARING_NOT_FOUND)
+            );
+
+        return casesMapper.mapToEvents(eventRepository.findAllByHearingId(hearing.getId()));
```

Comment:
I'm not entirely sure why we need the case_id param here if we choose a hearing_id as once we have a hearing, we will know the single case. We should probably simplify the service here.
FYI @hemantasharma1129

Reply:
Agree, we dont need case_id as inout param.